src/controllers/groups/__tests__/utils.test.js

```diff
@@ -32,4 +32,16 @@ describe("Groups - UTILS", () => {
       message: "Group(s) not found",
     });
   });
+
+  it("should validate group names", () => {
+    expect(isGroupNameValid("Example_Word-123")).toBe(true);
+    expect(isGroupNameValid("__valid_underscores__")).toBe(true);
+    expect(isGroupNameValid("Valid---words")).toBe(true);
+    expect(isGroupNameValid("123InvalidStart")).toBe(false);
```

Comment:
```suggestion
    expect(isGroupNameValid("invalid?middle")).toBe(false);
    expect(isGroupNameValid("Valid-middle+7:options.work_ok")).toBe(true);
```
I think we should test all of the options in the reg-ex.

Reply:
Ash's (AI) suggestions are far better than mine